src/targets/gpu/lowering.cpp

```diff
@@ -542,37 +542,23 @@ struct miopen_apply
                 new_inputs);
         });
 
-        apply_map.emplace("gpu::concat_past_present", [=](instruction_ref ins) {
-            return mod->replace_instruction(
-                ins,
-                make_op("gpu::precompile_op", {{"op", to_value(ins->get_operator())}}),
-                ins->inputs());
-        });
-
-        apply_map.emplace("gpu::compute_attention_probabilities", [=](instruction_ref ins) {
-            auto s          = ins->get_shape();
-            auto output     = insert_allocation(ins, s);
-            auto new_inputs = ins->inputs();
-            new_inputs.push_back(output);
-            return mod->replace_instruction(
-                ins,
-                make_op("gpu::precompile_op", {{"op", to_value(ins->get_operator())}}),
-                new_inputs);
+        apply_map.emplace("gpu::concat_past_present_k", [=](instruction_ref ins) {
+            return mod->replace_instruction(ins,
+                                            make_op("gpu::precompile_op",
+                                                    {{"op", to_value(ins->get_operator())},
+                                                     {"output_shape", to_value(ins->get_shape())}}),
+                                            ins->inputs());
         });
 
-        apply_map.emplace("gpu::gqa_softmax", [=](instruction_ref ins) {
-            auto s      = ins->get_shape();
-            auto inputs = ins->inputs();
-
-            auto new_inputs = ins->inputs();
-            new_inputs.push_back(inputs.at(2));
-            return mod->replace_instruction(
-                ins,
-                make_op("gpu::precompile_op", {{"op", to_value(ins->get_operator())}}),
-                new_inputs);
+        apply_map.emplace("gpu::concat_past_present_v", [=](instruction_ref ins) {
+            return mod->replace_instruction(ins,
+                                            make_op("gpu::precompile_op",
+                                                    {{"op", to_value(ins->get_operator())},
+                                                     {"output_shape", to_value(ins->get_shape())}}),
+                                            ins->inputs());
         });
 
-        apply_map.emplace("gpu::compute_attention_scores", [=](instruction_ref ins) {
+        apply_map.emplace("gpu::kv_cache_attention", [=](instruction_ref ins) {
```

Comment:
So this will add a `precompile_op` using `gpu::kv_cache_attention` but there is no jit kernel for `gpu::kv_cache_attention`, which is why there is an exception. 

I thought this was supposed to use MLIR and not a custom kernel?